x/erc20/keeper/ibc_callbacks.go

```diff
@@ -89,6 +89,12 @@ func (k Keeper) OnRecvPacket(
 		data.Denom, data.Amount,
 	)
 
+	// If the coin denom starts with `factory/` then it is a token factory coin, and we should not convert it
+	// NOTE: Check https://docs.osmosis.zone/osmosis-core/modules/tokenfactory/ for more information
+	if strings.HasPrefix(data.Denom, "factory/") {
```

Comment:
What if I create a malicious chain, connect IBC to Evmos and spam trash, randomly denom to Evmos?

Reply:
There won't even be a need for a malicious chain, any legitimate chain like Osmosis that is already connected with Evmos, can spam unlimited factory Coins. This is why they are being ignored for now. We are trying to figure out how to register legitimate factory Coins.

Reply:
I meant, we are trying to prevent spamming coins. But this one only prevent spamming from Osmosis or legitimate chains. But how about preventing spamming coins from malicious chains? Is this considered?

Reply:
Any `tokenfactory` coin is handled by this check from any chain that supports them. For any malicious chain scenario we cannot be certain at the protocol level if a new chain is malicious and it's up to the community to flag them as malicious and not relay their IBC transfers to us.